app/scripts/controllers/registrationModal.js

```diff
@@ -131,5 +133,43 @@ angular
           validateRegistrant.blockVisible(block, registrant, true, conference)
         );
       };
+
+      $scope.nonAllowedTypeKeys = ['SPOUSE'];
+      $scope.excludeNonAllowedType = function (type) {
+        if (type.defaultTypeKey === 'SPOUSE') {
+          const coupleTypes = _.filter(conference.registrantTypes, {
+            defaultTypeKey: 'COUPLE',
+          });
+
+          if (coupleTypes.length === 0) {
+            return true;
+          }
+
+          // Check if this spouse is associated with any couple
+          const spouseIsAssociated = coupleTypes.some((coupleType) => {
+            if (
+              !coupleType.allowedRegistrantTypeSet ||
+              coupleType.allowedRegistrantTypeSet.length === 0
+            ) {
+              return false;
+            }
+
+            const hasAssociation = coupleType.allowedRegistrantTypeSet.some(
+              (association) => {
+                const isMatch = association.childRegistrantTypeId === type.id;
+
+                const hasChildRegistrants =
+                  association.numberOfChildRegistrants > 0;
```

Comment:
[Must-fix] Logic assumes numberOfChildRegistrants exists without validation. Evidence: `association.numberOfChildRegistrants > 0`. Impact: Runtime error if property is undefined. Fix: Add validation `association.numberOfChildRegistrants && association.numberOfChildRegistrants > 0`. [Category: File]